pkg/component/worker/kubeproxy_windows.go

```diff
@@ -0,0 +1,61 @@
+package worker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/k0sproject/k0s/pkg/assets"
+	"github.com/k0sproject/k0s/pkg/constant"
+	"github.com/k0sproject/k0s/pkg/supervisor"
+)
+
+type KubeProxy struct {
+	K0sVars    constant.CfgVars
+	LogLevel   string
+	supervisor supervisor.Supervisor
+}
+
+// Init
+func (k KubeProxy) Init() error {
+	return assets.Stage(k.K0sVars.BinDir, "kube-proxy.exe", constant.BinDirMode)
+}
+
+func (k KubeProxy) Run() error {
+	node, err := getNodeName()
```

Comment:
Do we need to always override the hostname? If not, I'd propose to do e.g. an env var to override it and thus "externalize" the problem out of k0s

Reply:
We need to override it for EC2 instances and I think it's better to have in the code, because it could not that user friendly to understand which EC2 hostname to use, because calico docs states that we need to use internal hostname always, so that could led user to have some confusion.